### ISSUE TITLE
Lint/Syntax may be reporting incorrect error

### ISSUE THREAD

**Issue by wilg** [rubocop-hq/rubocop#6178]
On this snippet:

```rb
  takes_block :arg, :class => Whatever do
    also_takes_block(:arg) do |a, b|
      # whatever
    end
  end
```

Rubocop reports:

```
Lint/Syntax: unexpected token kDO_BLOCK
(Using Ruby 2.3 parser; configure using TargetRubyVersion parameter, under AllCops)
```

As far as I know this is valid Ruby.

This error did not occur in 0.54.0.

I'm not sure how to work around this since Lint/Syntax cannot be disabled.

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
› bundle exec rubocop -V
0.58.2 (using Parser 2.5.1.2, running on jruby 2.3.3 java)
```

**Comment by wilg** [comment]
I worked around this by adding this to my `Gemfile`:

```rb
gem 'parser', '< 2.5.1.0'
```

**Comment by Drenmi** [comment]
Hi @wilg! The `Lint/Syntax` cop simply wraps diagnostic errors from the `parser` gem. Could you please open an issue with [whitequark/parser](https://github.com/whitequark/parser)?

**Comment by wilg** [comment]
Sure! https://github.com/whitequark/parser/issues/525
